internal/pkg/agent/application/actions/handlers/handler_action_diagnostics.go

```diff
@@ -0,0 +1,204 @@
+		var wBuf bytes.Buffer
+		defer func() {
+			if str := wBuf.String(); str != "" {
+				h.log.Warn(str)
+			}
+		}()
+		err := diagnostics.ZipArchive(&wBuf, &b, aDiag, uDiag)
+		if err != nil {
+			action.Err = err
+			return fmt.Errorf("error creating diagnostics bundle: %w", err)
+		}
+		r = &b
+		s = int64(b.Len())
+	} else {
+		defer func() {
+			f.Close()
+			os.Remove(f.Name())
+		}()
+		r = f
+	}
+	h.log.Debug("Sending diagnostics archive.")
+	uploadID, err := h.uploader.UploadDiagnostics(ctx, action.ActionID, ts.Format("2006-01-02T15-04-05Z07-00"), s, r) // RFC3339 format that uses - instead of : so it works on Windows
+	action.Err = err
+	action.UploadID = uploadID
+	if err != nil {
+		return fmt.Errorf("unable to upload diagnostics: %w", err)
+	}
+	h.log.Debugf("Diagnostics action '%+v' complete.", a)
+	return nil
+}
+
+func (h *Diagnostics) runHooks(ctx context.Context) ([]client.DiagnosticFileResult, error) {
+	hooks := append(h.coord.DiagnosticHooks(), diagnostics.GlobalHooks()...)
+	diags := make([]client.DiagnosticFileResult, 0, len(hooks))
+	for _, hook := range hooks {
+		if ctx.Err() != nil {
+			return diags, ctx.Err()
+		}
+		diags = append(diags, client.DiagnosticFileResult{
+			Name:        hook.Name,
+			Filename:    hook.Filename,
+			Description: hook.Description,
+			ContentType: hook.ContentType,
+			Content:     hook.Hook(ctx),
+			Generated:   time.Now().UTC(),
+		})
+	}
+	return diags, nil
+}
+
+func (h *Diagnostics) diagUnits(ctx context.Context) []client.DiagnosticUnitResult {
+	uDiag := make([]client.DiagnosticUnitResult, 0)
+	rr := h.coord.PerformDiagnostics(ctx)
+	for _, r := range rr {
+		diag := client.DiagnosticUnitResult{
+			ComponentID: r.Component.ID,
+			UnitID:      r.Unit.ID,
+			UnitType:    cproto.UnitType(r.Unit.Type),
+		}
+		if r.Err != nil {
+			diag.Err = r.Err
+		} else {
+			results := make([]client.DiagnosticFileResult, 0, len(r.Results))
+			for _, res := range r.Results {
+				results = append(results, client.DiagnosticFileResult{
+					Name:        res.Name,
+					Filename:    res.Filename,
+					Description: res.Description,
+					ContentType: res.ContentType,
+					Content:     res.Content,
+					Generated:   res.Generated.AsTime(),
+				})
+			}
+			diag.Results = results
+		}
+		uDiag = append(uDiag, diag)
+	}
+	return uDiag
+}
+
+// diagFile will write the diagnostics to a temporary file and return the file ready to be read
+func (h *Diagnostics) diagFile(aDiag []client.DiagnosticFileResult, uDiag []client.DiagnosticUnitResult) (*os.File, int64, error) {
+	f, err := os.CreateTemp("", "elastic-agent-diagnostics")
+	if err != nil {
+		return nil, 0, err
+	}
+
+	name := f.Name()
+	var wBuf bytes.Buffer
+	defer func() {
+		if str := wBuf.String(); str != "" {
+			h.log.Warn(str)
+		}
+	}()
+	if err := diagnostics.ZipArchive(&wBuf, f, aDiag, uDiag); err != nil {
+		os.Remove(name)
+		return nil, 0, err
+	}
+	f.Sync()
```

Comment:
Be great to hide this lint by just changing it to:

```suggestion
	_ = f.Sync()
```